docs/source/format/CanonicalExtensions.rst

```diff
@@ -72,4 +72,26 @@ same rules as laid out above, and provide backwards compatibility guarantees.
 Official List
 =============
 
-No canonical extension types have been standardized yet.
+Fixed size tensor
+=================
+
+* Extension name: `arrow.fixed_size_tensor`.
```

Comment:
One more thought about the naming: in the discussions (eg https://github.com/apache/arrow/issues/15483), we typically spoke about a fixed or constant _shape_ (and a fixed size can still mean varying shapes). 
So I wonder if a more correct name would be "fixed shape tensor" ?

Reply:
That would be better, yes.

Reply:
As we don't use this anywhere yet and "shape" is more precise I'd prefer it too.
(Varying shape issue is here: https://github.com/apache/arrow/issues/24868)